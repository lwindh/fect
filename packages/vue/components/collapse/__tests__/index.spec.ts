import { mount, flushPromises } from '@vue/test-utils'
import Collapse from '../index'
import CollapseGroup from '../../collapse-group'
import { later } from '../../../tests'

describe('Collapse', () => {
  it('should be render as a element', () => {
    const wrapper = mount(Collapse, {
      props: {
        title: 'test',
      },
    })
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should be support title and subtitle', () => {
    const wrapper = mount(Collapse, {
      props: {
        title: 'HTML',
        subtitle: 'HyperText Markup Language',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('h3').text()).toBe('HTML')
    expect(wrapper.find('.subtitle').text()).toBe('HyperText Markup Language')
  })
  it('should be support subTag and shadow mode', () => {
    const wrapper = mount(Collapse, {
      props: {
        title: 'HTML',
        subtitle: 'HyperText Markup Language',
        subTag: 'code',
        shadow: true,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('code')).toBeTruthy()
    expect(wrapper.find('.fect-collapse--shadow')).toBeTruthy()
  })

  it('should be tigger click', async () => {
    const wrapper = mount(Collapse, {
      props: {
        title: 'HTML',
        visible: false,
      },
    })

    await wrapper.find('.fect-collapse__view').trigger('click')
    expect(wrapper.find('.fect-collapse__expand').attributes('style')).toBe('visibility: hidden; height: 0px;')
  })
  it('should be support accordion model', async () => {
    const wrapper = mount({
      components: {
        [Collapse.name]: Collapse,
        [CollapseGroup.name]: CollapseGroup,
      },
      data() {
        return { show: [0], accordion: true }
      },
      template: `
      <div class="container">
        <fe-collapseGroup v-model="show" :accordion="accordion">
          <fe-collapse title="Test">
            123456
          </fe-collapse>
          <fe-collapse title="Test1">
            1
          </fe-collapse>
        </fe-collapseGroup>
      </div>
      `,
    })
    expect(wrapper.html()).toMatchSnapshot()
    const els = wrapper.findAll('.fect-collapse__view')
    await els[0].trigger('click')
    await flushPromises()
    await later()
    expect(wrapper.vm.show.length).toBe(0)
    await els[0].trigger('click')
    await flushPromises()
    await later()
    await els[1].trigger('click')
    expect(wrapper.vm.show).toEqual([1])
    await wrapper.setData({ accordion: false })
    await els[0].trigger('click')
    expect(wrapper.vm.show).toEqual([1, 0])
  })
})
