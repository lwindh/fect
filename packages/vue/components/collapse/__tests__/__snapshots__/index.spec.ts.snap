// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse should be support accordion model 1`] = `
<div class="container">
  <div class="fect-collapse__group">
    <div class="fect-collapse">
      <div class="fect-collapse__view" role="collapseButton">
        <div class="title">
          <h3>Test</h3><svg class="fect-collapse__svg--active" viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision" style="color: currentColor;">
            <path d="M6 9l6 6 6-6"></path>
          </svg>
        </div>
        <!---->
      </div>
      <div class="fect-collapse__expand" style="visibility: visible; height: auto;">
        <div class="fect-collapse__content"> 123456 </div>
      </div>
    </div>
    <div class="fect-collapse">
      <div class="fect-collapse__view" role="collapseButton">
        <div class="title">
          <h3>Test1</h3><svg class="fect-collapse__svg" viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision" style="color: currentColor;">
            <path d="M6 9l6 6 6-6"></path>
          </svg>
        </div>
        <!---->
      </div>
      <div class="fect-collapse__expand" style="visibility: hidden; height: 0px;">
        <div class="fect-collapse__content"> 1 </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse should be support subTag and shadow mode 1`] = `
<div class="fect-collapse--shadow">
  <div class="fect-collapse__view" role="collapseButton">
    <div class="title">
      <h3>HTML</h3><svg class="fect-collapse__svg" viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision" style="color: currentColor;">
        <path d="M6 9l6 6 6-6"></path>
      </svg>
    </div>
    <div class="subtitle"><code>HyperText Markup Language</code></div>
  </div>
  <div class="fect-collapse__expand" style="visibility: hidden; height: 0px;">
    <div class="fect-collapse__content">
      <!---->
    </div>
  </div>
</div>
`;

exports[`Collapse should be support title and subtitle 1`] = `
<div class="fect-collapse">
  <div class="fect-collapse__view" role="collapseButton">
    <div class="title">
      <h3>HTML</h3><svg class="fect-collapse__svg" viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision" style="color: currentColor;">
        <path d="M6 9l6 6 6-6"></path>
      </svg>
    </div>
    <div class="subtitle">HyperText Markup Language</div>
  </div>
  <div class="fect-collapse__expand" style="visibility: hidden; height: 0px;">
    <div class="fect-collapse__content">
      <!---->
    </div>
  </div>
</div>
`;
